genson/src/main/java/com/owlike/genson/reflect/EvolvableHandler.java

```diff
@@ -0,0 +1,63 @@
+package com.owlike.genson.reflect;
+
+import com.owlike.genson.Context;
+import com.owlike.genson.Converter;
+import com.owlike.genson.ext.jsr353.JSR353Bundle;
+import com.owlike.genson.stream.ObjectReader;
+import com.owlike.genson.stream.ObjectWriter;
+
+import javax.json.JsonValue;
+
+import java.util.Map;
+import java.util.function.Consumer;
+
+/**
+ * An implementation of an {@link UnknownPropertyHandler} that supports
+ * evolution of data classes via {@link Evolvable} interface.
+ * <p>
+ * If the target object we are deserializing into is {@link Evolvable},
+ * this handler will add any unknown properties encountered during
+ * deserialization into {@link Evolvable#unknownProperties()} map,
+ * and will write them out along with all known properties during
+ * subsequent serialization.
+ * <p>
+ * This prevents data loss when serializing and deserializing the same
+ * JSON payload using different versions of Java data classes.
+ *
+ * @author Aleksandar Seovic  2018.05.20
+ */
+public class EvolvableHandler implements UnknownPropertyHandler {
+    private static final Converter<JsonValue> CONVERTER = new JSR353Bundle.JsonValueConverter();
+
+    @Override
+    public <T> Consumer<T> readUnknownProperty(String propName, ObjectReader reader, Context ctx) {
```

Comment:
No, it would make it impossible to pass the bean we are deserializing into to `Consumer.accept` method within `BeanDescriptor`:
```
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /Users/aseovic/Projects/oss/genson/genson/src/main/java/com/owlike/genson/reflect/BeanDescriptor.java:[129,82] incompatible types: T cannot be converted to capture#1 of ?
[ERROR] /Users/aseovic/Projects/oss/genson/genson/src/main/java/com/owlike/genson/reflect/BeanDescriptor.java:[193,59] incompatible types: T cannot be converted to capture#2 of ?
[INFO] 2 errors 
```